test/functional/ui/screen.lua

```diff
@@ -422,18 +511,31 @@ asynchronous (feed(), nvim_input()) and synchronous API calls.
   - wait() can trigger redraws and consequently generate more indeterminism.
     In that case try removing every wait().
       ]])
+    did_warn = true
+  end
+
+
+  if err then
+    assert(false, err)
+  elseif did_warn then
     local tb = debug.traceback()
     local index = string.find(tb, '\n%s*%[C]')
     print(string.sub(tb,1,index))
   end
 
-  if err then
-    assert(false, err)
+  if flags.intermediate then
+    assert(intermediate_seen, "intermediate state expected before success")
```

Comment:
it's helpful to put "expected" at the start. Also mentioning "success" might be confusing in the context of an error.

    assert(intermediate_seen, "expected intermediate screen state, but got final screen state")